libgalois/include/galois/graphs/LS_LC_CSR_Graph.h

```diff
@@ -106,11 +107,21 @@ class LS_LC_CSR_Graph : private boost::noncopyable {
   // Compute the prefix sum using the two level method
   void computePrefixSum() {
     // todo: switch to parallel prefix sum when `galois::PrefixSum` is fixed
-    std::transform_inclusive_scan(
-        m_vertices.begin(), m_vertices.end(), m_pfx_sum_cache.begin(),
-        std::plus<uint64_t>(),
-        [](VertexMetadata const& v) { return v.degree(); }, 0ul);
-    m_prefix_valid.store(true, std::memory_order_release);
+    // todo: compiler does not apparently support the full C++17 standard
+    // std::transform_inclusive_scan(
+    //     m_vertices.begin(), m_vertices.end(), m_pfx_sum_cache.begin(),
+    //     std::plus<uint64_t>(),
+    //     [](VertexMetadata const& v) { return v.degree(); }, 0ul);
+
+    // sad, hand-rolled sequential prefix sum
+
+    m_prefix_valid.store(true, std::memory_order_relaxed);
```

Comment:
If this is truly needed then it needs to be set at the end with a memory barrier. I suspect nothing is happening in parallel on the machine with this. I am not sure why this is being replaced with something sequential, that could really hurt ingestion performance of large graphs.

Reply:
We should look at how it is being used more closely because there is no synchronization to prevent multiple invocations of `computePrefixSum` as-is.

For now, I'd like to keep the sequential implementation to go ahead with the other performance improvements and not break @divija95's stuff.

Reply:
Restored the parallel prefix sum implementation.